spec/requests/endpoints/discovery_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe "Discovery endpoint" do
+  before do
+    default_scopes_exist :read
+    optional_scopes_exist :write, :publish
+  end
+
+  it "returns json" do
+    get "/.well-known/oauth-authorization-server"
+
+    response_status_should_be(200)
+
+    expect(json_response).to have_key("issuer")
+    expect(json_response).to have_key("authorization_endpoint")
+    expect(json_response).to have_key("token_endpoint")
+    expect(json_response).to have_key("revocation_endpoint")
+    expect(json_response).to have_key("userinfo_endpoint")
+    expect(json_response).to have_key("scopes_supported")
+    expect(json_response).to have_key("response_types_supported")
+    expect(json_response).to have_key("response_modes_supported")
+    expect(json_response).to have_key("grant_types_supported")
+    expect(json_response).to have_key("token_endpoint_auth_methods_supported")
+    expect(json_response).to have_key("code_challenge_methods_supported")
+
+    expect(json_response["issuer"]).to be_a(String)
+
+    expect(json_response["scopes_supported"]).to be_a(Array)
+    expect(json_response["response_types_supported"]).to be_a(Array)
+    expect(json_response["response_modes_supported"]).to be_a(Array)
+    expect(json_response["grant_types_supported"]).to be_a(Array)
+    expect(json_response["token_endpoint_auth_methods_supported"]).to be_a(Array)
+    expect(json_response["code_challenge_methods_supported"]).to be_a(Array)
```

Comment:
I'm not sure if I should really test logic of the discovery response here, so instead I'm just asserting properties and types for things that must be arrays.

Reply:
Very similar to what we did for a one-off in our application:

```
      get oauth_discovery_path

      expect(response).to have_http_status(200)

      expect(response.content_type).to eq("application/json; charset=utf-8")

      object = JSON.parse(response.body)

      expect(object.dig("authorization_endpoint")).to eq("http://www.example.com/oauth/authorize")
      expect(object.dig("token_endpoint")).to eq("http://www.example.com/oauth/token")
      expect(object.dig("revocation_endpoint")).to eq("http://www.example.com/oauth/revoke")
      ...
```

Reply:
Yeah, similar to the Mastodon one as well: https://github.com/mastodon/mastodon/blob/main/spec/requests/well_known/oauth_metadata_spec.rb

I'm just not sure I want to be validating the values or not here, since the paths can be configured with doorkeeper.